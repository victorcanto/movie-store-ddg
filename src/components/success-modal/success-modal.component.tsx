import React, { useEffect, useState } from 'react';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Modal from '@mui/material/Modal';
import { Link } from 'react-router-dom';
import {
	selectIsCompletePurchase,
	selectUserNameFromCheckout,
	setCartItems,
	setFavoriteItems,
	setIsCompletedPurchase,
	useAppDispatch,
	useAppSelector,
} from '../../store';
import { Button } from '../button/button.component';

const style = {
	position: 'absolute' as 'absolute',
	top: '50%',
	left: '50%',
	transform: 'translate(-50%, -50%)',
	width: 400,
	bgcolor: 'background.paper',
	color: 'black',
	boxShadow: 3,
	p: 4,
};

export function SuccessModal() {
	const [open, setOpen] = useState(false);
	const handleOpen = () => setOpen(true);
	const handleClose = () => setOpen(false);

	const dispatch = useAppDispatch();

	const isCompletePurchase = useAppSelector(selectIsCompletePurchase);
	const userNameFromCheckout = useAppSelector(selectUserNameFromCheckout);

	const initialPageHandler = () => {
		dispatch(setCartItems([]));
		dispatch(setFavoriteItems([]));
		dispatch(setIsCompletedPurchase(false));
	};

	useEffect(() => {
		if (isCompletePurchase) {
			handleOpen();
		}
	}, [isCompletePurchase]);

	return (
		<div>
			<Modal
				data-testid='success-modal'
				open={open}
				onClose={handleClose}
				aria-labelledby='modal-modal-title'
				aria-describedby='modal-modal-description'
			>
				<Box sx={style}>
					<Typography
						textAlign='center'
						id='modal-modal-title'
						variant='h3'
					>
						Obrigado
						<Typography textAlign='center' variant='h4'>
							{userNameFromCheckout}!
						</Typography>
					</Typography>
					<Typography
						id='modal-modal-description'
						sx={{ mt: 2, mb: 2, textAlign: 'center' }}
					>
						Sua compra foi realizada com sucesso!
					</Typography>
					<Button
						component={Link}
						to='/'
						onClick={initialPageHandler}
					>
						Ir para loja
					</Button>
				</Box>
			</Modal>
		</div>
	);
}
