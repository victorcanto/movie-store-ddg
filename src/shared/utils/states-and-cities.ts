import statesAndCities from '../data/estados-cidades.json';

export const getBrazilianStates = () =>
	statesAndCities.estados.map(({ nome }) => nome);

export const getBrazilianCitiesByState = (stateName: string) =>
	statesAndCities.estados.find(({ nome }) => nome === stateName)?.cidades;
