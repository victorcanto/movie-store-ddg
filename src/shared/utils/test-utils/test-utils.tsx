import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { setupStore, AppStore, RootState } from '../../../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {},
) {
	function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
		return (
			<Provider store={store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	}
	return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export { renderWithProviders as render };
