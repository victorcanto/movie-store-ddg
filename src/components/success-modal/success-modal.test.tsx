import React from 'react';
import { render } from '../../shared/utils/test-utils';
import { SuccessModal } from './success-modal.component';

test('snapshot of the base state', () => {
	const { asFragment } = render(<SuccessModal />);
	expect(asFragment()).toMatchSnapshot();
});
