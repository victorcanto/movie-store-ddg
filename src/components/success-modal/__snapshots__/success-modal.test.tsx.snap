// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
